py/server/deephaven/updateby.py

```diff
@@ -267,3 +266,85 @@ def forward_fill(cols: Union[str, List[str]]) -> UpdateByOperation:
+        fwd_ticks (int): the look-forward window size (int rows/ticks), default is 0
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        return UpdateByOperation(j_updateby_op=_JUpdateByOperation.RollingSum(rev_ticks, fwd_ticks, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a rolling sum (tick) UpdateByOperation.") from e
+
+
+def rolling_sum_time(ts_col: str, cols: Union[str, List[str]], rev_time: Union[int, str],
+                     fwd_time: Union[int, str] = 0) -> UpdateByOperation:
+    """Creates a rolling sum UpdateByOperation for the supplied column names, using time as the windowing unit. This 
+    function accepts nanoseconds as the reverse and forward window parameters. Negative values are allowed and can be 
+    used to generate completely forward or completely reverse windows. A row containing a null in the timestamp 
+    column belongs to no window and will not have a value computed or be considered in the windows of other rows. 
+     
+    Here are some examples of window values:
+        rev_time = 0, fwd_time = 0 - contains rows that exactly match the current row timestamp
+        rev_time = 600_000_000_000, fwd_time = 0 - contains rows from 10m earlier through the current row timestamp (
+            inclusive)
+        rev_time = 0, fwd_time = 600_000_000_000 - contains rows from the current row through 10m following the
+            current row timestamp (inclusive)
+        rev_time = 600_000_000_000, fwd_time = 600_000_000_000 - contains rows from 10m earlier through 10m following
+            the current row timestamp (inclusive)
+        rev_time = 600_000_000_000, fwd_time = -30_000_000_000 - contains rows from 10m earlier through 5m before the
+            current * row timestamp (inclusive), this is a purely backwards looking window
+        rev_time = -300_000_000_000, fwd_time = 600_000_000_000} - contains rows from 5m following through 10m
+            following the  current row timestamp (inclusive), this is a purely forwards looking window
+    
+    Args:
+        ts_col (str):
+        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output,
+            i.e. "new_col = col"; when empty, update_by perform the forward fill operation on all columns.
+        rev_time (int): the look-behind window size, can be expressed as an integer in nanoseconds or a time
+            interval string, e.g. "00:00:00.001"
+        fwd_time (int): the look-ahead window size, can be expressed as an integer in nanoseconds or a time
+            interval string, e.g. "00:00:00.001", default is 0
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        rev_time = _JDateTimeUtils.expressionToNanos(rev_time) if isinstance(rev_time, str) else rev_time
```

Comment:
Was curious about specifying negative values when providing strings to `DateTimeUtils.expressionToNanos()`.  I produced this test which passes and we may want to add to `TestDateTimeUtils.java` since we will be relying on proper production of positive/negative time durations in UpdateBy rolling operators.

```
    public void testExpresionToNanos() {
        final long SECOND_NANOS = 1_000_000_000;
        final long MINUTE_NANOS = 60 * SECOND_NANOS;
        final long HOUR_NANOS = 60 * MINUTE_NANOS;

        // second fractions
        assertEquals(1_000_000L, DateTimeUtils.expressionToNanos("00:00:00.001"));
        assertEquals(-1_000_000L, DateTimeUtils.expressionToNanos("-00:00:00.001"));
        assertEquals(100_000_000L, DateTimeUtils.expressionToNanos("00:00:00.1"));
        assertEquals(-100_000_000L, DateTimeUtils.expressionToNanos("-00:00:00.1"));

        // seconds
        assertEquals(1 * SECOND_NANOS, DateTimeUtils.expressionToNanos("00:00:01.00"));
        assertEquals(-1 * SECOND_NANOS, DateTimeUtils.expressionToNanos("-00:00:01.00"));
        assertEquals(11 * SECOND_NANOS, DateTimeUtils.expressionToNanos("00:00:11.00"));
        assertEquals(-11 * SECOND_NANOS, DateTimeUtils.expressionToNanos("-00:00:11.00"));
        // non-standard but functional (> 60 seconds)
        assertEquals(150 * SECOND_NANOS, DateTimeUtils.expressionToNanos("00:00:150.00"));
        assertEquals(-150 * SECOND_NANOS, DateTimeUtils.expressionToNanos("-00:00:150.00"));

        // minutes
        assertEquals(1 * MINUTE_NANOS, DateTimeUtils.expressionToNanos("00:01:00"));
        assertEquals(-1 * MINUTE_NANOS, DateTimeUtils.expressionToNanos("-00:01:00"));
        assertEquals(11 * MINUTE_NANOS, DateTimeUtils.expressionToNanos("00:11:00"));
        assertEquals(-11 * MINUTE_NANOS, DateTimeUtils.expressionToNanos("-00:11:00"));
        // non-standard but functional (> 60 minutes)
        assertEquals(150 * MINUTE_NANOS, DateTimeUtils.expressionToNanos("00:150:00"));
        assertEquals(-150 * MINUTE_NANOS, DateTimeUtils.expressionToNanos("-00:150:00"));

        // hours
        assertEquals(1 * HOUR_NANOS, DateTimeUtils.expressionToNanos("01:00:00"));
        assertEquals(-1 * HOUR_NANOS, DateTimeUtils.expressionToNanos("-01:00:00"));
        assertEquals(11 * HOUR_NANOS, DateTimeUtils.expressionToNanos("11:00:00"));
        assertEquals(-11 * HOUR_NANOS, DateTimeUtils.expressionToNanos("-11:00:00"));

        // days are multiples of hours
        assertEquals(24 * HOUR_NANOS, DateTimeUtils.expressionToNanos("24:00:00"));
        assertEquals(-24 * HOUR_NANOS, DateTimeUtils.expressionToNanos("-24:00:00"));

        // weeks are multiples of hours
        assertEquals(168 * HOUR_NANOS, DateTimeUtils.expressionToNanos("168:00:00"));
        assertEquals(-168 * HOUR_NANOS, DateTimeUtils.expressionToNanos("-168:00:00"));

        // days
        assertEquals(100_000 * HOUR_NANOS, DateTimeUtils.expressionToNanos("100000:00:00"));
        assertEquals(-100_000 * HOUR_NANOS, DateTimeUtils.expressionToNanos("-100000:00:00"));
    }
```